journal_rec_components.py

```diff
@@ -0,0 +1,143 @@
+import base64
+import json
+
+import dash
+import numpy as np
+import pandas as pd
+import requests
+
+from journal_rec_ml_models import get_neighbors
+from journal_rec_word_vectors import parse_uploaded_file
+
+def displayTapNodeData(data):
+    """
+    This function is designed to render the data table
+    for the user
+    
+    Args:
+        data - the node that the user has highlighted
+    """
+    if data and 'journal' in data:
+        if data['data_type'] == 'paper':
+            columns=[
+                {"name":col.upper(), "id":col} 
+                for col in ["journal", "paper"]
+            ]
+            data=[
+                {
+                    "journal":data['journal'],
+                    "paper":data['label']
+                }
+            ]
+        else:
+            columns=[
+                {"name":col.upper(), "id":col} 
+                for col in ["journal"]
+            ]
+            data = [
+                {"journal":data['journal']}
+            ]
+        return columns, data
+    else:
+        columns = [{"name":"journal", "id":"journal"}]
+        data = [{"journal":"Joural Title Goes here"}]
+        return columns, data
+    
```

Comment:
Not sure whether you want to follow PEP8 strictly. If yes, you need two blank lines here.